lang/rust/avro/src/schema.rs

```diff
@@ -1282,6 +1280,23 @@ impl Parser {
         Ok(schema)
     }
 
+    fn validate_name(
+        &self,
+        name: &String
+    ) -> bool {
+        let i1 = name.len();
+        if i1 == 0 {
+            return false;
+        }
+        let mut result : bool = name.chars().nth(0).unwrap().is_alphabetic();
+
+        let size = name.chars().count();
+        for i in 1..size {
+            result &= name.chars().nth(i).unwrap().is_alphanumeric();
+        }
+        return result;
```

Comment:
Actually I think you can simplify the last 6 lines to `name.chars().any(|c| !c.is_alphabetic())` (implicit return)

Reply:
how to distinguish first and others ? need to have a kind of "indexedAny" ?

Reply:
would that work `name.chars().drop(1).all(|c| c.is_alphabetic())` ?
I haven't tried it

Reply:
I finally found :
`name.char_indices().all(|(index, c)| ...)`